### PR TITLE
Splash Screen for Android

### PR DESCRIPTION
**Pull request by saketkumar95**
Splash screen for android, doing it native.
Supports all devices.

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -12,26 +12,30 @@
...
-      android:label="@string/app_name"
-      android:icon="@mipmap/ic_launcher"
-      android:theme="@style/AppTheme">
-      <activity
-        android:name=".MainActivity"
+        android:name=".MainApplication"
+        android:allowBackup="true"
         android:label="@string/app_name"
-        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
-        android:windowSoftInputMode="adjustResize">
```
Comment (vishwesh3) [inline android/app/src/main/AndroidManifest.xml]: Removing this has introduced layout issue.
When keyboard pops up navbar is removed from the screen.
screen shot 2017-06-15 at 10 10 49 am

Comment (saketkumar95) [inline android/app/src/main/AndroidManifest.xml]: oh ! Missed this. Thanks @vishwesh3 👍

**Comment by saketkumar95** [comment]
>There is again a loading screen, which is shown when background process is going on. So this as introduced a brand new 2 second delay.

I think this delay is not in iOS app?

>We would like to share code between android and iOS as much as possible. (so at latter stage if we decide to change design it becomes super easy)

For iOS we don't need to add any code. We can just add the designs for splash screen.
Check out : https://medium.com/the-react-native-log/change-default-launch-screen-in-react-native-ios-app-544f94f1e947

**Comment by borisyankov** [comment]
I think we have a splash screen on iOS... not 100% sure we are showing it.

In any case, a potential splash screen needs to:
 * not increase the loading time, but show it over the 'loading progress' thing we have
 * be the same between iOS and Android
 * be consistent with the web app

Currently, the image doesn't look like anything we have on the web app.
We should be consolidating the 'logo' and not creating more versions.

I tested about 10 chat apps right now, and most of them have no splash screen and no progress bar even. Best case for us is to speedup loading and just show the UI itself.

**Comment by saketkumar95** [comment]
Here is the demo:
splash

**Comment by saketkumar95** [comment]
@borisyankov Is this fine? I added logo instead of having loading ( inspired by airbnb app) Shoul i resolve the merge conflicts or close it?

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (borisyankov) [reply android/app/src/main/AndroidManifest.xml]: Would you fix the code to have only the needed lines produce a diff (no indentation where none is needed)? It is hard to determine what was added.

**Comment by borisyankov** [comment]
Closing as no longer needed.

**Pull request closed without merging**